llvm/test/CodeGen/X86/yk-o0-fixstackmap-bug.ll

```diff
@@ -0,0 +1,252 @@
+!53 = !DILocation(line: 19, column: 9, scope: !38)
+!54 = !DILocation(line: 19, column: 14, scope: !38)
+!55 = !DILocation(line: 20, column: 27, scope: !38)
+!56 = !DILocation(line: 20, column: 3, scope: !38)
+!57 = !DILocalVariable(name: "loc", scope: !38, file: !3, line: 21, type: !58)
+!58 = !DIDerivedType(tag: DW_TAG_typedef, name: "YkLocation", file: !51, line: 22, baseType: !59)
+!59 = distinct !DICompositeType(tag: DW_TAG_structure_type, file: !51, line: 20, size: 64, elements: !60)
+!60 = !{!61}
+!61 = !DIDerivedType(tag: DW_TAG_member, name: "state", scope: !59, file: !51, line: 21, baseType: !62, size: 64)
+!62 = !DIDerivedType(tag: DW_TAG_typedef, name: "uintptr_t", file: !63, line: 90, baseType: !64)
+!63 = !DIFile(filename: "/usr/include/stdint.h", directory: "", checksumkind: CSK_MD5, checksum: "a48e64edacc5b19f56c99745232c963c")
+!64 = !DIBasicType(name: "unsigned long", size: 64, encoding: DW_ATE_unsigned)
+!65 = !DILocation(line: 21, column: 14, scope: !38)
+!66 = !DILocation(line: 21, column: 20, scope: !38)
+!67 = !DILocation(line: 23, column: 3, scope: !38)
+!68 = !{i64 2147816749}
+!69 = !DILocation(line: 24, column: 3, scope: !38)
+!70 = !{i64 2147816795}
+!71 = !DILocalVariable(name: "result", scope: !38, file: !3, line: 26, type: !11)
+!72 = !DILocation(line: 26, column: 8, scope: !38)
+!73 = !DILocalVariable(name: "i", scope: !74, file: !3, line: 27, type: !41)
+!74 = distinct !DILexicalBlock(scope: !38, file: !3, line: 27, column: 3)
+!75 = !DILocation(line: 27, column: 12, scope: !74)
+!76 = !DILocation(line: 27, column: 8, scope: !74)
+!77 = !DILocation(line: 27, column: 19, scope: !78)
+!78 = distinct !DILexicalBlock(scope: !74, file: !3, line: 27, column: 3)
+!79 = !DILocation(line: 27, column: 21, scope: !78)
+!80 = !DILocation(line: 27, column: 3, scope: !74)
+!81 = !DILocation(line: 28, column: 21, scope: !82)
+!82 = distinct !DILexicalBlock(scope: !78, file: !3, line: 27, column: 32)
+!83 = !DILocation(line: 28, column: 23, scope: !82)
+!84 = !DILocation(line: 28, column: 19, scope: !82)
+!85 = !DILocation(line: 29, column: 25, scope: !82)
+!86 = !DILocation(line: 29, column: 5, scope: !82)
+!87 = !DILocation(line: 30, column: 14, scope: !82)
+!88 = !DILocation(line: 30, column: 12, scope: !82)
+!89 = !DILocation(line: 31, column: 45, scope: !82)
+!90 = !DILocation(line: 31, column: 48, scope: !82)
+!91 = !DILocation(line: 31, column: 5, scope: !82)
+!92 = !DILocation(line: 32, column: 3, scope: !82)
+!93 = !DILocation(line: 27, column: 28, scope: !78)
+!94 = !DILocation(line: 27, column: 3, scope: !78)
+!95 = distinct !{!95, !80, !96, !97}
+!96 = !DILocation(line: 32, column: 3, scope: !74)
+!97 = !{!"llvm.loop.mustprogress"}
+!98 = !DILocation(line: 34, column: 3, scope: !38)
+!99 = !DILocation(line: 35, column: 14, scope: !38)
+!100 = !DILocation(line: 35, column: 3, scope: !38)
+!101 = !DILocation(line: 37, column: 3, scope: !38)
+
+^0 = module: (path: "[Regular LTO]", hash: (0, 0, 0, 0, 0))
```

Comment:
Is this stuff required for the test to work? I've never seen this before in any module dump.

Reply:
yes, if I remove this the test fails.

Reply:
I've also never seen this syntax!

Reply:
https://llvm.org/docs/LangRef.html#module-path-summary-entry

It's to do with Thin LTO, where you use module summaries. We don't use thin LTO, but I guess it bleeds across a bit somehow.